firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteMutationQueue.java

```diff
@@ -249,6 +249,16 @@ public MutationBatch getNextMutationBatchAfterBatchId(int batchId) {
         .firstValue(row -> decodeInlineMutationBatch(row.getInt(0), row.getBlob(1)));
   }
 
+  @Override
+  public int getLargestUnacknowledgedBatchId() {
+    if (isEmpty()) {
```

Comment:
Nit: In case the mutation queue is not empty, you are running a very similar query twice.

Reply:
Fixed.